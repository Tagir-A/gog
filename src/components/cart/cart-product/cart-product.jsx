import React, { Component } from "react";
import "./cart-product.css";

class CartProduct extends Component {
  render() {
    const { product } = this.props;
    return (
      <div className="cart__product">
        <img src={product.img} alt={product.title} className="product__img" />
        <div className="product__info">
          <h4 className="product__info__game-title">{product.title}</h4>
          <button
            className="product__info__btn"
            onClick={this.handleRemoveClick}
          >
            Remove
          </button>
        </div>
        <h6 className="product__price">
          {product.price.value / product.price.minorUnits}
        </h6>
      </div>
    );
  }

  handleRemoveClick = () => {
    const { onRemoveClick, product } = this.props;
    onRemoveClick && onRemoveClick(product);
  };
}

export { CartProduct };
