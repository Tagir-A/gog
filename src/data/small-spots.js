import oddworld from '../assets/small-spots/oddworld.png'
import deponia from '../assets/small-spots/deponia.png'
import theSettlers2 from '../assets/small-spots/the-settlers-2.png'
import neverwinterNights from '../assets/small-spots/neverwinter-nights.png'
import assasinsCreed from '../assets/small-spots/assasins-creed.png'

const SMALL_SPOTS = [{
        title: "ODDWORLD: STRANGER'S WRATH",
        img: oddworld,
        price: {
            currency: 'USD',
            value: 999,
            minorUnits: 100
        },
        discount: 50
    },
    {
        title: "CHAOS ON DEPONIA",
        img: deponia,
        price: {
            currency: 'USD',
            value: 999,
            minorUnits: 100
        },
        owned: true
    },
    {
        title: "THE SETTLERS 2: GOLD EDITION",
        img: theSettlers2,
        price: {
            currency: 'USD',
            value: 599,
            minorUnits: 100
        }
    },
    {
        title: "NEVERWINTER NIGHTS",
        img: neverwinterNights,
        price: {
            currency: 'USD',
            value: 499,
            minorUnits: 100
        },
        discount: 50
    },
    {
        title: "ASSASIN'S CREED: DIRECTOR'S CUT",
        img: assasinsCreed,
        price: {
            currency: 'USD',
            value: 999,
            minorUnits: 100
        }
    }
]

export {
    SMALL_SPOTS
}
