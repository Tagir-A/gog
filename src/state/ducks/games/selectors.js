export function isInCart(game, {
    products
}) {
    return Boolean(products.find(product => product.title === game.title));
}
