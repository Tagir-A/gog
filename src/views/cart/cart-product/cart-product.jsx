import React, { Component } from "react";
import PropTypes from "prop-types";
import "./cart-product.css";
import game from "../../../models/game";

class CartProduct extends Component {
  static propTypes = {
    product: game.isRequired,
    onRemoveClick: PropTypes.func.isRequired
  };
  render() {
    const { product } = this.props;
    return (
      <div className="cart__product">
        <img src={product.img} alt={product.title} className="product__img" />
        <div className="product__info">
          <h4 className="product__info__game-title">{product.title}</h4>
          <button
            className="product__info__btn"
            onClick={this.handleRemoveClick}
          >
            Remove
          </button>
        </div>
        <h6 className="product__price">
          {product.price.value / product.price.minorUnits}
        </h6>
      </div>
    );
  }

  handleRemoveClick = () => {
    const { onRemoveClick, product } = this.props;
    onRemoveClick(product);
  };
}

export { CartProduct };
