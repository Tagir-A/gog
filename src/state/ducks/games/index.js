import reducer from "./reducers";
import * as gamesSelectors from "./selectors";

export {
    gamesSelectors
}
export default reducer;
