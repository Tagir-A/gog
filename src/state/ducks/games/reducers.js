import { GAMES } from "../../../data/games";

const defaultState = [...GAMES];

export default function games(state = defaultState, action = {}) {
  return state;
}
