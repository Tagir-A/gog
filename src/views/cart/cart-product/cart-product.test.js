import {GAMES} from '../../../data/games'
import { CartProduct } from './cart-product';

describe('cart-product', () => {
    it('renders as expected', () => {
        const wrapper = shallow(<CartProduct
            product={GAMES[0]}
            onRemoveClick={()=>null}
            />)
        expect(wrapper).toMatchSnapshot()
    })
})