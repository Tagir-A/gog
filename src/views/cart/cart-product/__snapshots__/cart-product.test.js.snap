// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cart-product renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CartProduct
    onRemoveClick={[Function]}
    product={
        Object {
            "discount": 50,
            "img": "oddworld.png",
            "price": Object {
              "currency": "USD",
              "minorUnits": 100,
              "value": 999,
            },
            "title": "ODDWORLD: STRANGER'S WRATH",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="ODDWORLD: STRANGER'S WRATH"
          className="product__img"
          src="oddworld.png"
/>,
        <div
          className="product__info"
>
          <h4
                    className="product__info__game-title"
          >
                    ODDWORLD: STRANGER'S WRATH
          </h4>
          <button
                    className="product__info__btn"
                    onClick={[Function]}
          >
                    Remove
          </button>
</div>,
        <h6
          className="product__price"
>
          9.99
</h6>,
      ],
      "className": "cart__product",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "ODDWORLD: STRANGER'S WRATH",
          "className": "product__img",
          "src": "oddworld.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4
              className="product__info__game-title"
>
              ODDWORLD: STRANGER'S WRATH
</h4>,
            <button
              className="product__info__btn"
              onClick={[Function]}
>
              Remove
</button>,
          ],
          "className": "product__info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ODDWORLD: STRANGER'S WRATH",
              "className": "product__info__game-title",
            },
            "ref": null,
            "rendered": "ODDWORLD: STRANGER'S WRATH",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Remove",
              "className": "product__info__btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Remove",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 9.99,
          "className": "product__price",
        },
        "ref": null,
        "rendered": 9.99,
        "type": "h6",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="ODDWORLD: STRANGER'S WRATH"
            className="product__img"
            src="oddworld.png"
/>,
          <div
            className="product__info"
>
            <h4
                        className="product__info__game-title"
            >
                        ODDWORLD: STRANGER'S WRATH
            </h4>
            <button
                        className="product__info__btn"
                        onClick={[Function]}
            >
                        Remove
            </button>
</div>,
          <h6
            className="product__price"
>
            9.99
</h6>,
        ],
        "className": "cart__product",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "ODDWORLD: STRANGER'S WRATH",
            "className": "product__img",
            "src": "oddworld.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4
                className="product__info__game-title"
>
                ODDWORLD: STRANGER'S WRATH
</h4>,
              <button
                className="product__info__btn"
                onClick={[Function]}
>
                Remove
</button>,
            ],
            "className": "product__info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ODDWORLD: STRANGER'S WRATH",
                "className": "product__info__game-title",
              },
              "ref": null,
              "rendered": "ODDWORLD: STRANGER'S WRATH",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Remove",
                "className": "product__info__btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Remove",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 9.99,
            "className": "product__price",
          },
          "ref": null,
          "rendered": 9.99,
          "type": "h6",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
